项目打包时，是否存在该问题

node_modules/element-plus/lib/el-tree/src/model/node.d.ts:28:24 - error TS2304: Cannot find name 'Nullable'.

28     get nextSibling(): Nullable<Node>;
                          ~~~~~~~~

node_modules/element-plus/lib/el-tree/src/model/node.d.ts:29:28 - error TS2304: Cannot find name 'Nullable'.

29     get previousSibling(): Nullable<Node>;
                              ~~~~~~~~

node_modules/resize-observer-polyfill/src/index.d.ts:19:18 - error TS2717: Subsequent property declarations must have the same type.  Property 'contentRect' must be of type 'DOMRectReadOnly', but here has type 'DOMRectReadOnly'.

19         readonly contentRect: DOMRectReadOnly;
                    ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:12696:14
    12696     readonly contentRect: DOMRectReadOnly;
                       ~~~~~~~~~~~
    'contentRect' was also declared here.


Found 3 errors.

node 16.0.0 环境

--- a/src/composables/form.ts
+++ b/src/composables/form.ts
@@ -129,19 +129,19 @@ export function useFormMenu(
   })
 }
 
-export function useFormMethods<T = UnknownObject>(
+export function useFormMethods(
   emit: (
     event: 'update:modelValue' | 'submit' | 'reset',
     ...args: unknown[]
   ) => void
 ): {
-  form: Ref<IFormExpose<T>>
+  form: Ref<IFormExpose>
   loading: Ref<boolean>
   upFormData: (value: unknown) => void
   submitForm: () => void
   resetForm: (reset?: boolean) => void
-} & IFormExpose<T> {
-  const form = shallowRef<IFormExpose<T>>({} as IFormExpose<T>)
+} & IFormExpose {
+  const form = shallowRef<IFormExpose>({} as IFormExpose)
   const { show, toggleShow } = useShow()
 
   function validate(callback?: IFormValidateCallback) {
@@ -152,12 +152,12 @@ export function useFormMethods<T = UnknownObject>(
     form.value.resetFields()
   }
 
-  function clearValidate(props?: MaybeArray<DeepKeyof<T>>) {
+  function clearValidate(props?: MaybeArray<string>) {
     form.value.clearValidate(props)
   }
 
   function validateField(
-    props: MaybeArray<DeepKeyof<T>>,
+    props: MaybeArray<string>,
     cb: IFormValidateFieldCallback
   ) {
     form.value.validateField(props, cb)

--- a/src/composables/table.ts
+++ b/src/composables/table.ts
@@ -14,7 +14,6 @@ import type {
   IPagination,
   UnknownObject,
   StringObject,
-  DeepKeyof,
   MaybeRef,
   MaybeArray,
 } from '../types/index'
@@ -111,7 +110,7 @@ export function useTableMethods<T = UnknownObject>(): {
     table.value.clearSort()
   }
 
-  function clearFilter(columnKeys?: MaybeArray<DeepKeyof<T>>) {
+  function clearFilter(columnKeys?: MaybeArray<string>) {
     table.value.clearFilter(columnKeys)
   }
 
@@ -119,7 +118,7 @@ export function useTableMethods<T = UnknownObject>(): {
     table.value.doLayout()
   }
 
-  function sort(prop: DeepKeyof<T>, order: string) {
+  function sort(prop: string, order: string) {
     table.value.sort(prop, order)
   }
 

--- a/src/types/crud.ts
+++ b/src/types/crud.ts
@@ -105,4 +105,4 @@ export type ICrudSubmit = (
   invalidFields?: UnknownObject
 ) => void
 
-export type ICrudExpose<T = UnknownObject> = IFormExpose<T> & ITableExpose<T>
+export type ICrudExpose<T = UnknownObject> = IFormExpose & ITableExpose<T>

--- a/src/types/form.ts
+++ b/src/types/form.ts
@@ -106,16 +106,16 @@ export interface IFormValidateFieldCallback {
 }
 
 /** Form Expose Methods */
-export interface IFormExpose<T = StringObject> {
+export interface IFormExpose {
   /** validate the whole form. Takes a callback as a param. After validation, the callback will be executed with two params: a boolean indicating if the validation has passed, and an object containing all fields that fail the validation. Returns a promise if callback is omitted */
   validate: (callback?: IFormValidateCallback) => Promise<boolean>
   /** reset all the fields and remove validation result */
   resetFields: () => void
   /** clear validation message for certain fields. The parameter is prop name or an array of prop names of the form items whose validation messages will be removed. When omitted, all fields' validation messages will be cleared */
-  clearValidate: (props?: MaybeArray<DeepKeyof<T>>) => void
+  clearValidate: (props?: MaybeArray<string>) => void
   /** validate one or several form items */
   validateField: (
-    props: MaybeArray<DeepKeyof<T>>,
+    props: MaybeArray<string>,
     cb: IFormValidateFieldCallback
   ) => void
 }

--- a/src/types/table.ts
+++ b/src/types/table.ts
@@ -124,11 +124,11 @@ export interface ITableExpose<T = StringObject> {
   /** clear sorting, restore data to the original order */
   clearSort: () => void
   /** clear filters of the columns whose columnKey are passed in. If no params, clear all filters */
-  clearFilter: (columnKeys?: MaybeArray<DeepKeyof<T>>) => void
+  clearFilter: (columnKeys?: MaybeArray<string>) => void
   /** refresh the layout of Table. When the visibility of Table changes, you may need to call this method to get a correct layout */
   doLayout: () => void
   /** sort Table manually. Property prop is used to set sort column, property order is used to set sort order */
-  sort: (prop: DeepKeyof<T>, order: string) => void
+  sort: (prop: string, order: string) => void
 }
 
 /** Pagination Attributes */

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,6 +9,7 @@
     "declaration": true,
     "emitDeclarationOnly": true,
     "esModuleInterop": true,
+    "skipLibCheck": true,
     "declarationDir": "types",
     "lib": ["ESNext", "DOM"],
     "paths": {

